crypto/object_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crypto
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/crypto/keys"
+	"github.com/google/trillian/testonly"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/kylelemons/godebug/pretty"
+)
+
+// toFromBytes serializes obj using proto.Marshal, and returns the unmarshaled copy.
+func toFromBytes(obj proto.Message) (proto.Message, error) {
+	b, err := proto.Marshal(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	obj2 := obj
```

Comment:
obj has to be a pointer, otherwise `Reset()` wouldn't work, so `obj2 := obj; obj2.Reset()` ends up being the same as `obj.Reset()`.

You should also note in the docs that this method modifies the input, which I wouldn't expect from the name.